code/game/objects/items/weapons/blades.dm

```diff
@@ -20,6 +20,130 @@
+	traits += "Bicaridine - heal your target"
+
+	. += jointext(traits, "<br>")
+
+/obj/item/weapon/claymore/harvester/Initialize()
+	. = .. ()
+	beaker = new /obj/item/reagent_containers/glass/beaker/vial
+
+/obj/item/weapon/claymore/harvester/attackby(obj/item/I, mob/user)
+	if(user.action_busy)
+		return
+
+	if(!istype(I, /obj/item/reagent_containers))
+		return
+
+	var/trans
+	var/obj/item/reagent_containers/container = I
+	if(istype(container, /obj/item/reagent_containers/syringe))
+		trans = beaker.reagents.trans_to(container, 40)
+		to_chat(user, "<span class='rose'>You take [trans]u out of the internal storage. It now contains [beaker.reagents.total_volume]u.</span>")
+		return
+
+	if(length(container.reagents.reagent_list) > 1)
+		to_chat(user, "<span class='rose'>The solution needs to be uniform and contain only a single type of reagent to be compatible.</span>")
+		return FALSE
+
+	if(!locate(container.reagents.reagent_list[1].type) in loadable_reagents)
+		to_chat(user, "<span class='rose'>This reagent is not compatible with the weapon's mechanism. Check the engraved symbols for further information.</span>")
+		return FALSE
+
+	if(container.reagents.total_volume < 10)
+		to_chat(user, "<span class='rose'>At least 10u of the substance is needed.</span>")
+		return FALSE
+
+	trans = container.reagents.trans_to(beaker, 40)
+	to_chat(user, "<span class='rose'>You load [trans]u into the internal system. It now holds [beaker.reagents.total_volume]u.</span>")
+
+/obj/item/weapon/claymore/harvester/unique_action(mob/user)
+	if(effect_active)
+		to_chat(user, "<span class='rose'>The blade is powered with [beaker.reagents.reagent_list[1]]. You can release the effect by stabbing a creature.</span>")
+		return
+
+	if(effect_loaded)
+		effect_active = TRUE
+		to_chat(user, "<span class='rose'>You release the holding mechanism, empowering your next attack.</span>")
+		return TRUE
+
+	if(beaker.reagents.total_volume < 10)
+		to_chat(user, "<span class='rose'>You don't have enough substance.</span>")
+		return FALSE
+
+	if(user.action_busy)
+		return
+
+	to_chat(user, "<span class='rose'>You start filling up the small chambers along the blade's edge.</span>")
+	if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_BAR, ignore_turf_checks = TRUE))
+		to_chat(user, "<span class='rose'>You are interrupted! The safety machanism drains out the reagent back into the main storage.</span>")
+		return FALSE
+
+	effect_loaded = TRUE
+	return TRUE
+
+/obj/item/weapon/claymore/harvester/attack(mob/living/M, mob/living/user)
+	if(effect_active)
+		switch(beaker.reagents.reagent_list[1].type)
+			if(/datum/reagent/medicine/tramadol)
+				M.apply_status_effect(/datum/status_effect/incapacitating/slowdown, 2 SECONDS)
+
+			if(/datum/reagent/medicine/kelotane)
+				var/turf/target = get_turf(M)
+				target.ignite()
+				var/list/cone_turfs = generate_cone(user, 2, 1, 91, Get_Angle(user, M.loc))
+				for(var/turf/T in cone_turfs)
```

Comment:
For loop that type checks when it shouldnt

Reply:
I get "code\game\objects\items\weapons\blades.dm:131:error: T.ignite: undefined proc" when I just use var/T instead of var/turf/T.

Reply:
changed